### PR TITLE
[DEV-96] Adding logic to delete projects

### PR DESCRIPTION
**Pull request by cash1998**
- [x] Added a column called "is_deleted" to denote a project's soft delete
- [x] Updated existing project fetches to fetch only non-deleted projects
- [x] Added logic to restore projects if a project is added again via webhook
- [x] Added logic to delete projects using Github app/webhook 
- [x] Testing on local for these changes

### CODE REVIEW

**Review by dhirenmathur — CHANGES_REQUESTED** [review]

File: server/projects.py [hunk 1]
```diff
@@ -303,7 +316,7 @@ def get_parsed_project_branches(self, repo_name, user_id, default):
             # Build the base query
             query = (
                 "SELECT id, branch_name, repo_name, updated_at, is_default,"
-                " status FROM projects WHERE user_id = %s"
+                f" status FROM projects WHERE user_id = %s {self.is_deleted_condition}"
             )
```
[the remark this answers is not in the document]
Comment (cash1998) [reply server/projects.py:320]: This is to increase readability- a line shoudn't have more than 120 characters.

[the remark this answers is not in the document]
Comment (dhirenmathur) [reply server/projects.py:320]: no i meant use multiline string

**Review by dhirenmathur — CHANGES_REQUESTED** [review]

File: server/projects.py [hunk 2]
```diff
@@ -69,7 +77,7 @@ def list_projects(self):
         try:
             conn = psycopg2.connect(os.getenv("POSTGRES_SERVER"))
             cursor = conn.cursor()
-            cursor.execute("SELECT id, directory, is_default FROM projects")
+            cursor.execute(f"SELECT id, directory, is_default FROM projects {self.is_deleted_condition}")
```
[the remark this answers is not in the document]
Comment (dhirenmathur) [reply server/projects.py:80]: also coincidently was reading something and found this
image

[the remark this answers is not in the document]
Comment (cash1998) [reply server/projects.py:80]: The sql injection will happen only if the string is a variable, and is an input from some place
In this case it's a hardcoded value

File: server/routers/webhook.py [hunk 3]
```diff
@@ -111,14 +115,24 @@ async def parse_repos(payload, request: Request):
...
-                })
+                }
             project_manager.update_project_status(
                 project_id, ProjectStatusEnum.READY
             )
             github.close()
+    elif "action" in payload and payload["action"] == "removed":
+        auth, installation_auth, user_details, github = GithubService.get_app_auth_details(payload)
+        user_id = user_details[0]
+        if repository_field in payload:
```
[the remark this answers is not in the document]
Comment (cash1998) [reply server/routers/webhook.py:132]: repository_field is - 
    "added": "repositories_added",
        "created": "repositories"
        "created": "repositories",
        "removed": "repositories_removed"
    }.get(payload["action"], None)
    
    fetching the value from the dictionary on the basis of "action" value

**Review by dhirenmathur — APPROVED** [review]

**Pull request merged**